### PR TITLE
feat: add generate report implementation

### PR DESCRIPTION
**Pull request by eivenlour**
**Changes:**
- Added the generate report implementation. All we have to do now is connect it to the UI stuff (button/popup dialog).

**Sample usage:**
```
generateReport(
   'https://example.com',
   [categories.PERFORMANCE, categories.ACCESSIBILITY],
   deviceForms.DESKTOP
);
```
**Todo:**
- Validate URLs. We can use regex here.
- Handle errors.

### CODE REVIEW

File: src/commands/report.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const aws = require('aws-sdk');
+const lighthouse = require('lighthouse');
+const chromeLauncher = require('chrome-launcher');
+const { v1: uuidv1 } = require('uuid');
+const config = require('../config');
+// const { categories, deviceForms } = require('../constants');
```
Comment (HelenGezahegn) [inline src/commands/report.js]: Comment, and is that the automatic indentation? So thicc LOL.

File: src/commands/report.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	secretAccessKey: config('AWS_SECRET_ACCESS_KEY'),
+	region: config('REGION')
+});
+
+const uploadFile = file => {
+	const uniqueId = uuidv1();
+	const fileName = `report-${uniqueId}.html`;
+	const fileType = 'text/html';
+	const storageClass = 'STANDARD';
+	const acl = 'public-read'
```
Comment (eivenlour) [inline src/commands/report.js]: These tab indentations are bothering me! I swear on my IDE, I'm using 2-space indentations... does this happen for everyone too?

File: src/constants.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+'use strict'
+
+const categories = {
+  ACCESSIBILITY: 'accessibility',
+  BEST_PRACTICES: 'best-practices',
+	PERFORMANCE: 'performance',
```
Comment (HelenGezahegn) [inline src/constants.js]: Indentation

**Review by HelenGezahegn — APPROVED** [review]
Other than the styling changes, LGTM. 🚀

File: src/commands/report.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	};
+
+	s3.upload(params, function(err, data) {
+		if (err) {
+			console.log('Something went wrong. Try again later.');
+		}
+			console.log(`File uploaded successfully. ${data.Location}`);
+	});
+
+};
```
Comment (eivenlour) [inline src/commands/report.js]: Still have to handle this error in the future when the report file fails to upload in the S3 bucket.

File: src/commands/report.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		emulatedFormFactor: deviceForm,
+		port: chrome.port
+	};
+	const runnerResult = await lighthouse(url, options);
+
+	const reportFile = Buffer.from(runnerResult.report, 'utf-8');
+	uploadFile(reportFile);
+
+  await chrome.kill();
+};
```
Comment (eivenlour) [inline src/commands/report.js]: This function takes in:
- a URL: still needs to be validated and make sure it's a working URL first. Else, don't generate report.
- a list of categories: should be based on the checked boxes from the popup dialog.
- a device form: should be based on the radio button choices from the popup dialog.

File: src/commands/report.js (hunk 2 above)
Comment (HelenGezahegn) [reply src/commands/report.js]: LOL nah sis. 😂

**Pull request merged**